test/integration/base.py

```diff
@@ -0,0 +1,113 @@
+import contextlib
+import errno
+import os
+import socket
+import subprocess
+import tempfile
+import unittest
+
+
+class PelikanTest(unittest.TestCase):
+    reserved_ports = set()
+
+    @staticmethod
+    def default_path():
+        return os.path.realpath(os.path.join(
+            os.path.dirname(os.path.abspath(__file__)),
+            '../../_build/_bin'
+        ))
+
+    # from http://stackoverflow.com/a/22106569
```

Comment:
Can you split out the socket related logic into its own class and module/file? This will allow multiple test files to share the same logic instead of having all tests in a giant file.